testnet/stacks-node/src/nakamoto_node/miner.rs

```diff
@@ -170,13 +171,35 @@ impl BlockMinerThread {
         )
         .expect("FATAL: could not open sortition DB");
         let mut sortition_handle = sort_db.index_handle_at_tip();
+        let aggregate_public_key = if block.header.chain_length <= 1 {
+            signer.aggregate_public_key.clone()
+        } else {
+            let block_sn = SortitionDB::get_block_snapshot_consensus(
+                sortition_handle.conn(),
+                &block.header.consensus_hash,
+            )?
+            .ok_or(ChainstateError::DBError(DBError::NotFoundError))?;
+            let aggregate_key_block_header =
+                NakamotoChainState::get_canonical_block_header(chain_state.db(), &sort_db)?
+                    .unwrap();
+
+            let aggregate_public_key = NakamotoChainState::get_aggregate_public_key(
+                &sort_db,
+                &sortition_handle,
+                &mut chain_state,
+                block_sn.block_height,
+                &aggregate_key_block_header.index_block_hash(),
+            )?;
+            aggregate_public_key
+        };
```

Comment:
This should be de-duped with the invocation in the mockamoto node, perhaps a

`NakamotoChainState::get_stacker_key_for_block(&mut chain_state, &sortdb, &block.header)`

Or, perhaps this logic should be part of the invocation of `get_aggregate_public_key`, which should take `&block.header` instead of `block_sn.block_height` and `at_block: StacksBlockId`. Otherwise, we're going to have the logic of how to find an aggregate key for a given block repeated many times in the codebase (if we haven't already?)

Reply:
Yup, can do